Opentrons/buildroot
# Kernel update: add rv3028 RTC support

**This PR is not ready to be merged and should only be used for testing RTC for the refresh project.**

This PR adds support for the rv3028 RTC module by:
- updating the linux kernel to the one with latest drivers
- adding overlay for rv3028 from the kernel. The `dts/overlays/i2c-rtc.dts` file had to be compiled as 'IN TREE' and then moved to `/boot/overlays` as a `.dtbo` file since buildroot doesn't seem to have a way of fetching overlays specified in the kernel. The other way to fetch this (specific version of) i2c-rtc.dts would be to update the rpi-firmware package. (but that would update *all* the overlays in /boot/overlays)
- adding a udev rule to set system clock from RTC on startup

Closes #108 

## Packages added:
- VCDBG & i2ctools: Debugging tools (actually, vcdbg can be removed)
- hwclock: tool for accessing the hardware clock (read from/ write to RTC)

## To test:
1. Flash the built image to sd card 
2. Put it in an RPi that's connected to the rv3028 module
3. Once the system has booted, ssh into it and verify that the rtc has been registered: `dmesg | grep rtc`
4. If it has been registered, we will need to write the correct date to it first. The best way to do this is to update the system time (using either `date -s` or by letting the system update it by connecting to the internet), and then enter the command `hwclock -w`. This will write the time from system clock to RTC. **Make sure that the time saved is in UTC.**

That is it. Next time the robot boots, it'll fetch date&time from RTC during startup. You can check this by typing `timedatectl`. The system time and RTC time should be exactly same.

Troubleshooting tips:
1. If the RTC isn't responding, check if it is being recognized by the system by typing:
```
# modprobe i2c-dev
# i2cdetect -y 1
```
That should show something like this which shows that an i2c device of address 0x52 is being recognized and **U**sed
```
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: -- -- UU -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
#
```
If you don't see the above, there is probably some hardware problem
2. If the RTC is recognized and registered but you can't read from it using `timedatectl` or `hwclock`, try to write some valid value to it using `hwclock -w`. If there is an invalid value on it (which is usually true for a new device), then it might result in a read error.
